### ISSUE TITLE
How to use useQuery

### ISSUE THREAD

**Issue by JohannesKlauss** [PabloSzx/gqless-hooks#3]
I have a question because the `useQuery` isn't that clear to me.

Let's say I have a query defined as `findUserByName(name: String!): User`.

How would I trigger a query depending on a state variable? For example a user want to search for a user and types in an input field. Every time the state variable for the user input changes I want to trigger the query.
How would I set this up? The example seems to define the arguments inside the useQuery hook, but doesn't seem to define deps to retrigger the query building when the state variable updates.

**Comment by PabloSzx** [comment]
that's a good point, right now you would have to make a useEffect and calling the useQuery callback everytime the variable change, of course it's not the best, so I will have to make a 'variables' options and call the query automatically when it changes.

**Comment by JohannesKlauss** [comment]
With `useMutation` it seems to be the same issue.

**Comment by PabloSzx** [comment]
technically the mutation shouldn't need that feature, because the mutations should work as a response of an action, not like a query. It's that way in [useMutation of apollo](https://www.apollographql.com/docs/react/data/mutations/#usemutation-api)

**Comment by JohannesKlauss** [comment]
But wouldn't one create a mutation like this?
```ts

  const [createUser, createdUserData] = useMutation(async ({createUser}) => {
    const {_id} = await createUser({
      data: {
        name: nameFromState,
        password: passwordFromState
      }
    });

    return {_id: _id};
  });
```

**Comment by PabloSzx** [comment]
The thing is that createUser that comes from gqless is not an async function, is just an "accessor", and if it has arguments, works as a plain function that receives the arguments, its the same way for mutation and queries, and that's why I made this hook, because it wraps up the gqless queries and mutation in an async-like hook, in your example `const [createUser, ..`. is itself a promise, so the usage would be 

 ```tsx
   const [createUser, createdUserData] = useMutation(async ({createUser}) => {
     const {_id} = createUser({
       data: {
         name: nameFromState,
         password: passwordFromState
       }
     });
 
     return {_id: _id};
   });

 // ...
   <button  onClick={async () => {
  const data = await  createUser();
// "data" and the promise state is also available inside createdUserData 
}} >createUser</button>
 ```

**Comment by PabloSzx** [comment]
I will improve the documentation and more examples when I have more time, no worries about that :+1:

**Comment by PabloSzx** [comment]
I have been working on this project and there is quite some progress, I published v1.0.1 which adds a couple of features and some documentation

https://pabloszx.github.io/gqless-hooks

There is still pending some more advanced usage examples, but I hope the basic is already there.
